Tests/SwiftParserTest/Attributes.swift

```diff
@@ -17,4 +17,19 @@ final class AttributeTests: XCTestCase {
       ]
     )
   }
+  
+  func testMissingGenericTypeToAttribute() {
+    AssertParse(
+      """
+      @differentiable(reverse wrt#^DIAG_1^#,where T#^DIAG_2^##^DIAG_3^#
```

Comment:
You don’t need to specify two location markers here. You could just write `#^DIAG_2^#` here and use that marker for the second and third `DiagnosticSpec`.

Reply:
Good one. I've submitted another PR.